.github/workflows/main.yml

```diff
@@ -19,28 +18,23 @@ jobs:
       matrix:
         # Test all supported versions on Ubuntu:
         os: [ubuntu-latest]
-        python: ["3.7", "3.8", "3.9", "3.10", 3.11-dev, 3.12-dev]
+        # Skip 3.12-dev, pending https://github.com/python-greenlet/greenlet/issues/323
+        python: [pypy3.8, pypy3.9, "3.7", "3.8", "3.9", "3.10", 3.11-dev]
```

Comment:
If I'm understanding https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#handling-failures correctly, we can achieve what we want to with something like:

```yaml
    continue-on-error: ${{ matrix.failure-allowed }}
    strategy:
      fail-fast: false
      matrix:
        # Test all supported versions on Ubuntu:
        os: [ubuntu-latest]
        python: [pypy3.8, pypy3.9, "3.7", "3.8", "3.9", "3.10", 3.11-dev]
        failure-allowed: [false]
        include:
          # Also test macOS and Windows:
          - os: macos-latest
            python: "3.10"
            failure-allowed: false
          - os: windows-latest
            python: "3.10"
            failure-allowed: false
          # Also test in-development version, allowing failures
          - os: ubuntu-latest
            python: "3.12-dev"
            failure-allowed: true
```

Reply:
`failure-allowed` is a verb/phrase that we can change as we wish to communicate things via the variable name. The example from GitHub used `experimental`.

Reply:
Thanks, I did try something along those lines, but wasn't successful. 

Although I think https://github.com/python/pyperformance/pull/264 is a better approach? Then we'll at least see if the non-greenlet tests fail for 3.12. And when greenlet installs for 3.12, it'll start testing those too.

Reply:
Looking at your force-pushed events' diffs, I don't see `continue-on-error: ${{ matrix.failure-allowed }}` being added at any point at the `job.<id>` level; which might be why they didn't work?

Regardless, they're both valid approaches to solving the problem so it's not a big deal which one we go with. :)

Reply:
I also tried it in [another branch](https://github.com/hugovk/pyperformance/blob/ci-update2/.github/workflows/main.yml):

```yml
  test:
    runs-on: ${{ matrix.os }}
    name: ${{ matrix.os }} - ${{ matrix.python }}
    continue-on-error: ${{ matrix.experimental }}
    strategy:
      fail-fast: false
      matrix:
        # Test all supported versions on Ubuntu:
        os: [ubuntu-latest]
        python: [pypy3.7, pypy3.8, pypy3.9, "3.7", "3.8", "3.9", "3.10", 3.11-dev]
        experimental: [false]
        include:
          # Also test macOS and Windows:
          - os: macos-latest
            python: "3.10"
            experimental: false
          - os: windows-latest
            python: "3.10"
            experimental: false
          # Allow 3.12-dev to fail pending https://github.com/python-greenlet/greenlet/issues/323
          - os: ubuntu-latest
            python: 3.12-dev
            experimental: true
```

Very similar to your snippet:

<img width="1303" alt="image" src="https://user-images.githubusercontent.com/1324225/216832523-4eda6c68-76da-4c03-961c-0897a060b5c3.png">

But still failed a PR: https://github.com/hugovk/pyperformance/pull/3

Reply:
Indeed! Thanks for clarifying! ^.^